frontend/src/lib/Quiver.ts

```diff
@@ -715,6 +819,53 @@ but was expecting \`${JSON.stringify(expectedIndexTypes)}\`.
     return `${leftBracket + leftInterval}, ${rightInterval + rightBracket}`
   }
 
+  private static formatPeriodType(data: number, typeName: PeriodType): string {
+    const duration = Number(data)
+    const match = typeName.match(/period\[(.*)]/)
+    if (match === null) {
+      throw new Error(`Invalid period type: ${typeName}`)
+    }
+    const [, freq] = match
+    return this.formatPeriod(duration, freq)
+  }
+
+  private static formatPeriod(
+    duration: number,
+    freq: PeriodFrequency
+  ): string {
+    const [freqName, freqParam] = freq.split("-", 2)
+    const momentConverter =
+      PERIOD_TYPE_FORMATTERS[freqName as PandasOffsetType]
+    if (!momentConverter) {
+      throw new Error(`Unsupported period frequency: ${freq}`)
+    }
+    return momentConverter(duration, freqParam)
+  }
+
+  private static formatCategoricalType(
+    x: number | StructRow,
+    field: Field
+  ): string {
+    // Serialization for pandas.Interval and pandas.Period is provided by Arrow extensions
+    // https://github.com/pandas-dev/pandas/blob/235d9009b571c21b353ab215e1e675b1924ae55c/
+    // pandas/core/arrays/arrow/extension_types.py#L17
+    const extensionName = field.metadata.get("ARROW:extension:name")
+    if (extensionName) {
+      const extensionMetadata = JSON.parse(
+        field.metadata.get("ARROW:extension:metadata") as string
```

Comment:
Is `ARROW:extension:metadata` always expected to be set if there is a `ARROW:extension:name`?

Reply:
Yes. I have found no other case.